base/broadcast.jl

```diff
@@ -1261,4 +1261,36 @@ end
 end
 @inline broadcasted(::S, f, args...) where S<:BroadcastStyle = Broadcasted{S}(f, args)
 
+"""
+    BroadcastOp{F} <: Function
+
+Represents the "dotted" version of an operator, which broadcasts the operator over its
+arguments, so `BroadcastOp(op)` is functionally equivalent to `(x...) -> (op).(x...)`.
+
+Can be created by just passing an operator preceded by a dot to a higher-order function.
+
+# Examples
+```jldoctest
+julia> a = [[1 3; 2 4], [5 7; 6 8]]
+
+julia> b = [[9 11; 10 12], [13 15; 14 16]]
+
+julia> map(.*, a, b)
+2-element Array{Array{Int64,2},1}:
+ [9 33; 20 48]
+ [65 105; 84 128]
+
+julia> Base.BroadcastOp(+)(a, b) == a .+ b
+true
+```
+"""
+struct BroadcastOp{F} <: Function
+    f::F
+end
+
+@inline (op::BroadcastOp)(x...) = op.f.(x...)
+
+show(io::IO, op::BroadcastOp) = print(io, "Base.BroadcastOp(", op.f, ')')
```

Comment:
I think you can just do
```jl
print(io, BroadcastOp, '(', op.f, ')')
```
and it will automatically prepend the `Base.` if it `BroadcastOp` not exported.

Reply:
That prints for me as `Base.Broadcast.BroadcastOp`, which is probably fine as well

Reply:
I think we want to export it from the `Broadcast` module into `Base`?

Reply:
i.e. add an `export` line.

Reply:
Yes, it is already exported above, `print` just doesn't realize that

Reply:
This also seems to print the type parameter, so I think I will just leave it as it was originally

Reply:
Yes, it is already exported above, print just doesn't realize that